templates/table.js.tera

```diff
@@ -425,6 +425,23 @@ function colorizeHeaderRow{{ loop.index0 }}() {
 }
 {% endfor %}
 
+{% for row, l in header_ellipsis %}
+    function shortenHeaderRow{{ loop.index0 }}() {
+        var row = {{ row }};
+        var ellipsis = {{ l }};
+        var skip_label = {% if not header_labels | length == 0 %}true{% else %}false{% endif %};
+        $(`table > thead > tr:nth-child(${row + 1}) > td`).each(
+            function() {
+                value = this.innerHTML;
+                if (value.length > ellipsis && !skip_label) {
+                    this.innerHTML = `${value.substring(0, ellipsis)}<a tabindex="0" role="button" href="#" data-toggle="popover" data-trigger="focus" data-html='true' data-content='<div style="overflow: auto; max-height: 30vh; max-width: 25vw;">${value}</div>'>...</a>`;
+                }
+                skip_label = false;
+            }
+        );
+    }
```

Comment:
Is the code for the columns very similar? If so, I'd use a common function for that.

Reply:
Also, this looks like the function is rewritten for every header row. Why that? why not having a parameter in the function for the row? Ideally, all this should even work without the template code, just by checking for header rows with activated ellipsis in the config dict and applying the ellipsis for all of them.

Reply:
Only the one line is similar. I have moved the ellipsis information for the headers into the config. Now there's only a single function to apply it to all header rows.